metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/connector/parser/EmbeddedFilesResolver.java

```diff
@@ -122,13 +126,56 @@ public void process() throws IOException {
 			return;
 		}
 
+		// Parse the embedded files to find eventual new embedded files within them.
+		Collection<EmbeddedFile> temporaryEmbeddedFilesList = new ArrayList<>(processedEmbeddedFiles.values());
+		while (!temporaryEmbeddedFilesList.isEmpty()) {
+			// This is the list of new embedded files that we will find in the embedded files to process
+			final List<EmbeddedFile> newEmbeddedFiles = new ArrayList<>();
+			for (EmbeddedFile processedEmbeddedFile : temporaryEmbeddedFilesList) {
+				// Look for "${file::" pattern
+				final Matcher fileMatcher = FILE_PATTERN.matcher(processedEmbeddedFile.getContentAsString());
+				while (fileMatcher.find()) {
+					final String fileName = fileMatcher.group(1);
+
+					// If we encounter a new embedded file we process it, add it to the map of processed files
+					// and add it to the temporary list to process in the next loop iteration.
+					processedEmbeddedFiles.computeIfAbsent(
+						fileName,
+						name -> {
+							try {
+								final EmbeddedFile newEmbeddedFile = processFile(name, connectorDirectory);
+								newEmbeddedFiles.add(newEmbeddedFile);
+								return newEmbeddedFile;
+							} catch (Exception e) {
+								final String errorMessage = String.format(
+									"Error while processing file: %s. Current Connector directory: %s .",
+									name,
+									connectorDirectory
+								);
+								log.error(errorMessage, e);
+								throw new EmbeddedFileProcessingException(errorMessage, e);
+							}
+						}
+					);
+				}
+			}
+			temporaryEmbeddedFilesList = new ArrayList<>(newEmbeddedFiles);
+		}
+
 		JsonNodeUpdater
 			.jsonNodeUpdaterBuilder()
 			.withJsonNode(connector)
 			.withPredicate(value -> value.indexOf("${file::") != -1)
 			.withUpdater(this::performFileRefReplacements)
 			.build()
 			.update();
+
+		// Update the embedded files that are referencing other embedded files.
+		for (EmbeddedFile processedEmbeddedFile : processedEmbeddedFiles.values()) {
```

Comment:
Please modify the loop to iterate only over the embedded files that reference other embedded files, rather than looping through all of them. This will require creating a temporary list for those specific files.